samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/SectionContentsBean.java

```diff
@@ -510,6 +456,114 @@ public void setMetaData(SectionDataIfc section)
     this.timeLimit = value;
   }
 
+  public void setMetadataFixed(SectionDataIfc section) {
+
+	if (section.getSectionMetaDataByLabel(SectionDataIfc.NUM_QUESTIONS_FIXED) != null){
+		Integer numberfixed = new Integer(section.getSectionMetaDataByLabel(SectionDataIfc.NUM_QUESTIONS_FIXED));
+		setNumberToBeFixed(numberfixed);
+	}
+
+	if (section.getSectionMetaDataByLabel(SectionDataIfc.POOLID_FOR_FIXED_AND_RANDOM_DRAW) != null) {
+		Long poolid = new Long(section.getSectionMetaDataByLabel(SectionDataIfc.POOLID_FOR_FIXED_AND_RANDOM_DRAW));
+		setPoolIdToBeFixed(poolid);
+	}
+
+	if (section.getSectionMetaDataByLabel(SectionDataIfc.POOLNAME_FOR_FIXED_AND_RANDOM_DRAW) != null) {
```

Comment:
Just a random thought as this if and the previous one are separated. Is there a case where a poolname can be set and not the poolid (or viceversa)?

Reply:
In theory no. Or both or neither. Should I join the two if?

```
if (section.getSectionMetaDataByLabel(SectionDataIfc.POOLID_FOR_FIXED_AND_RANDOM_DRAW) != null) 
    &&  if (section.getSectionMetaDataByLabel(SectionDataIfc.POOLNAME_FOR_FIXED_AND_RANDOM_DRAW) != null) {
```

Reply:
Perhaps to make S2U-20 it is better to leave it as it is.

Reply:
Yep, fine as it is